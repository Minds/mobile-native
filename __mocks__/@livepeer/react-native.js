// mock @livepeer/react-native
export const LivepeerConfig = {
  setConfig: jest.fn(),
  getConfig: jest.fn(),
  get: jest.fn(),
  set: jest.fn(),
};

export const createReactClient = jest.fn();
export const createLivepeerClient = jest.fn();
export const Player = jest.fn();
