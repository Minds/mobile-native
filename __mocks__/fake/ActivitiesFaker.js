function load(count, container) {
  let activities = [...Array(count)].map((_, i) => {
    const code = 'activityguid' + i;
    return {
      attachment_guid: false,
      blurb: false,
      container_guid: code,
      custom_data: false,
      custom_type: false,
      rowKey: 'something' + i,
      description: 'Congratulations! ',
      edited: '',
      guid: code,
      mature: false,
      time_created: '1522036284',
      ownerObj: {
        guid: '824853017709780997',
        type: 'user',
        subtype: false,
        time_created: '1522036284',
      },
      message: 'Message',
      title: 'TITLE',
      owner_guid: '824853017709780997',
      parent_guid: '838106762591510528',
      perma_url: false,
      thumbnail_src: false,
      type: 'activity',
      wire_totals: {
        tokens: 1000000000000000000,
      },
      containerObj: container,
    };
  });

  return {
    activities: activities,
    'load-next': 'aaaaaa',
    'load-previous': 'aaaaaa',
  };
}

/**
 * comment service data faker
 * @param {integer} count
 */
export function activitiesServiceFaker() {
  return { load: load };
}

/**
 * Generate a single fake entity
 * @param {object} containerObj optional entity container
 * @returns
 */
export function fakeOne(containerObj) {
  const resp = load(1, containerObj);
  return resp.activities[0];
}
