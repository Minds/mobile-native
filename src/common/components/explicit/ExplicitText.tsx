import React, { Component } from 'react';

import { observer } from 'mobx-react';
import * as entities from 'entities';
import {
  Text,
  StyleSheet,
  View,
  Dimensions,
  Platform,
  TextStyle,
} from 'react-native';

import Tags from '../../../common/components/Tags';
import colors from '../../../styles/Colors';
import i18n from '../../services/i18n.service';
import ThemedStyles from '../../../styles/ThemedStyles';
import type ActivityModel from '../../../newsfeed/ActivityModel';
import { LinearGradient } from 'expo-linear-gradient';

type PropsType = {
  entity: ActivityModel;
  selectable?: boolean;
  navigation: any;
  style?: TextStyle | Array<TextStyle>;
  noTruncate?: boolean;
};

type StateType = {
  more: boolean;
  width: number;
  height: number;
};

@observer
export default class ExplicitText extends Component<PropsType, StateType> {
  needTruncate = false;

  state = {
    more: false,
    width: Dimensions.get('window').width,
    height: Dimensions.get('window').height,
  };

  /**
   * Handle dimensions changes
   */
  dimensionChange = () => {
    this.setState({
      width: Dimensions.get('window').width,
    });
  };

  /**
   * constructor
   */
  constructor(props) {
    super(props);
    Dimensions.addEventListener('change', this.dimensionChange);
  }

  /**
   * On component will unmount
   */
  componentWillUnmount() {
    // allways remove listeners on unmount
    Dimensions.removeEventListener('change', this.dimensionChange);
  }

  /**
   * Render
   */
  render() {
    const theme = ThemedStyles.style;
    const entity = this.props.entity;

    let title: string =
      entity.title && !entity.perma_url
        ? entities.decodeHTML(entity.title).trim()
        : '';
    let message = entity.text ? entities.decodeHTML(entity.text).trim() : '';

    if (title === message) {
      message = '';
    }

    let body: Element | null = null;
    let moreLess: JSX.Element | null = null;
    let explicitToggle = null;

    if (message !== '') {
      if (!this.props.noTruncate) {
        const truncated = this.truncate(message);
        // truncate if necessary
        if (message.length > truncated.length) {
          if (!this.state.more) message = truncated;
          moreLess = this.getMoreLess();
        }
      }

      body =
        entity.shouldBeBlured() && !entity.mature_visibility ? (
          <Text style={styles.mature}>{message}</Text>
        ) : (
          <Tags
            navigation={this.props.navigation}
            style={this.props.style}
            selectable={this.props.selectable}>
            {message}
          </Tags>
        );
    }

    const titleCmp: React.ReactNode = title ? (
      <Tags
        navigation={this.props.navigation}
        style={[
          this.props.style,
          theme.marginBottom,
          message ? styles.titleHasMessage : null,
        ]}>
        {title}
      </Tags>
    ) : null;

    return (
      <View
        style={[styles.container, !!entity.paywall ? styles.paywalled : null]}>
        {titleCmp}
        {body}
        {moreLess}
        {explicitToggle}
        {!!entity.paywall && this.renderGradient()}
      </View>
    );
  }

  renderGradient() {
    const backgroundColor = ThemedStyles.getColor('secondary_background');
    const startColor = backgroundColor + '00';
    const endColor = backgroundColor + 'FF';
    return (
      <LinearGradient colors={[startColor, endColor]} style={styles.linear} />
    );
  }

  /**
   * Truncate text
   */
  truncate(message: string) {
    const limit = this.getTextLimit();

    let truncated = false;
    let lines: Array<string> = message.split('\n');
    if (lines.length > 20) {
      lines = lines.slice(0, 20);
      truncated = true;
    }

    const strLines = lines.join('\n');
    if (strLines.length > limit) {
      return strLines.substr(0, limit) + '...';
    }
    return strLines + (truncated ? '...' : '');
  }

  /**
   * Get text char limit based on screen height
   */
  getTextLimit() {
    return this.state.height * 0.5;
  }

  /**
   * Returns more or less button
   */
  getMoreLess() {
    const msg = this.state.more ? i18n.t('showLess') : i18n.t('readMore');
    return (
      <Text style={styles.readmore} onPress={this.toggleReadMore}>
        {msg}
      </Text>
    );
  }

  /**
   * Toggle read more
   */
  toggleReadMore = () => {
    this.setState({ more: !this.state.more });
  };
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
  },
  paywalled: {
    maxHeight: 70,
    overflow: 'hidden',
  },
  linear: {
    position: 'absolute',
    height: 50,
    width: '100%',
    left: 0,
    bottom: 0,
    zIndex: 9999,
  },
  readmore: {
    color: colors.primary,
    marginTop: 5,
  },
  mature: {
    ...Platform.select({
      ios: {
        color: 'rgba(255, 255, 255, 0.1)',
        shadowColor: 'black',
        shadowOpacity: 1,
        shadowRadius: 2,
        shadowOffset: {
          height: -1,
        },
      },
      android: {
        color: 'transparent',
        textShadowColor: 'rgba(107, 107, 107, 0.5)',
        textShadowOffset: { width: -1, height: 1 },
        textShadowRadius: 20,
      },
    }),
  },
  titleHasMessage: {
    fontWeight: 'bold',
  },
});
