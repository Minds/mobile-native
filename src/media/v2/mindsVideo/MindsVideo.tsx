import React, { useEffect } from 'react';
import { observer, useLocalStore } from 'mobx-react';

import {
  TouchableWithoutFeedback,
  View,
  StyleProp,
  ViewStyle,
} from 'react-native';

import { ResizeMode, VideoReadyForDisplayEvent } from 'expo-av';
import videoPlayerService from '../../../common/services/video-player.service';
import type CommentModel from '../../../comments/CommentModel';
import type ActivityModel from '../../../newsfeed/ActivityModel';
import ThemedStyles from '../../../styles/ThemedStyles';
import createMindsVideoStore from './createMindsVideoStore';
import ExpoVideo from './Video';
import Error from './overlays/Error';
import Transcoding from './overlays/Transcoding';
import InProgress from './overlays/InProgress';
import Controls from './overlays/Controls';
import { useIsFocused } from '@react-navigation/native';

type Source = {
  src: string;
  size: number;
};

type PropsType = {
  entity?: ActivityModel | CommentModel;
  autoplay?: boolean;
  repeat?: boolean;
  resizeMode?: ResizeMode;
  video?: { uri: string };
  containerStyle?: StyleProp<ViewStyle>;
  onStoreCreated?: Function;
  onReadyForDisplay?: (event: VideoReadyForDisplayEvent) => void;
  hideOverlay?: boolean;
};

const MindsVideo = observer((props: PropsType) => {
  const theme = ThemedStyles.style;
  const localStore = useLocalStore(createMindsVideoStore, {
    entity: props.entity,
    autoplay: props.autoplay,
  });

  if (props.video && props.video.uri !== localStore.video.uri) {
    localStore.setVideo(props.video);
  }

  const onStoreCreated = props.onStoreCreated;

  const isFocused = useIsFocused();

  if (!isFocused && !localStore.paused) {
    localStore.pause();
  }

  useEffect(() => {
    onStoreCreated && onStoreCreated(localStore);
  }, [onStoreCreated, localStore]);

  // remove instance of video player service
  useEffect(() => {
    return () => {
      if (videoPlayerService.current === localStore) {
        videoPlayerService.clear();
      }
    };
  }, [localStore]);

  // Show inProgress overlay if load has started
  const inProgressOverlay = localStore.inProgress && <InProgress />;

  // Show Error overlay if not in progress and error
  const errorOverlay = !localStore.inProgress && localStore.error && <Error />;

  // Show Transcoding overlay if onError and isTranscoding Service -> true
  const transCodingOverlay = localStore.transcoding && <Transcoding />;

  // ^...else, show controls
  const controlsOverlay = !localStore.inProgress &&
    !localStore.error &&
    !localStore.transcoding && (
      <Controls
        localStore={localStore}
        entity={props.entity}
        hideOverlay={props.hideOverlay}
      />
    );

  return (
    <TouchableWithoutFeedback
      onPress={localStore.openControlOverlay}
      style={[theme.flexContainer, props.containerStyle]}>
      <View style={[theme.flexContainer, theme.backgroundBlack]}>
        <ExpoVideo
          entity={props.entity}
          localStore={localStore}
          repeat={props.repeat}
          resizeMode={props.resizeMode}
          onReadyForDisplay={props.onReadyForDisplay}
        />
        {inProgressOverlay}
        {errorOverlay}
        {transCodingOverlay}
        {controlsOverlay}
      </View>
    </TouchableWithoutFeedback>
  );
});

export default MindsVideo;
