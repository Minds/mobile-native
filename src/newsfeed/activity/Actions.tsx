import React from 'react';

import { StyleSheet, View, Platform } from 'react-native';

import { observer } from 'mobx-react';

import ThumbUpAction from './actions/ThumbUpAction';
import ThumbDownAction from './actions/ThumbDownAction';
import WireAction from './actions/WireAction';
import CommentsAction from './actions/CommentsAction';
import RemindAction from './actions/RemindAction';
import BoostAction from './actions/BoostAction';

import featuresService from '../../common/services/features.service';
import BaseModel from '../../common/BaseModel';
import type ActivityModel from '../ActivityModel';
import { useNavigation } from '@react-navigation/native';
import ThemedStyles from '../../styles/ThemedStyles';
import CommentsEntityOutlet from '../../comments/CommentsEntityOutlet';

type PropsType = {
  entity: ActivityModel;
  showCommentsOutlet?: boolean;
  onPressComment?: () => void;
};

export const Actions = observer((props: PropsType) => {
  const theme = ThemedStyles.style;
  const navigation = useNavigation();

  const entity = props.entity;
  const isOwner = entity.isOwner();
  const hasWire = Platform.OS !== 'ios';
  const hasCrypto = featuresService.has('crypto');
  const isScheduled = BaseModel.isScheduled(
    parseInt(entity.time_created, 10) * 1000,
  );

  return (
    <View>
      {entity && (
        <View style={[styles.container, theme.borderPrimary]}>
          <ThumbUpAction entity={entity} />
          <ThumbDownAction entity={entity} />
          <CommentsAction
            entity={entity}
            navigation={navigation}
            onPressComment={props.onPressComment}
            testID={
              props.entity.text === 'e2eTest' ? 'ActivityCommentButton' : ''
            }
          />
          <RemindAction entity={entity} />

          {!isOwner && hasCrypto && hasWire && (
            <WireAction owner={entity.ownerObj} navigation={navigation} />
          )}

          {isOwner && hasCrypto && !isScheduled && (
            <BoostAction entity={entity} navigation={navigation} />
          )}
        </View>
      )}
      {props.showCommentsOutlet ? (
        <CommentsEntityOutlet entity={entity} />
      ) : undefined}
    </View>
  );
});

export default Actions;

const styles = StyleSheet.create({
  container: {
    display: 'flex',
    flexDirection: 'row',
    // alignItems: 'center',
    justifyContent: 'space-evenly',
    paddingHorizontal: 10,
    // backgroundColor: 'blue',
    borderTopWidth: StyleSheet.hairlineWidth,
  },
  avatar: {
    height: 46,
    width: 46,
    borderRadius: 23,
    borderWidth: StyleSheet.hairlineWidth,
    borderColor: '#EEE',
  },
});
