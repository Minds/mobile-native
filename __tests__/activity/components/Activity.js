import 'react-native';
import React from 'react';
import { Text, TouchableOpacity } from 'react-native';
import { shallow } from 'enzyme';

import Activity from '../../../src/newsfeed/activity/Activity';

import { activitiesServiceFaker } from '../../../__mocks__/fake/ActivitiesFaker';

import ExplicitText from '../../../src/common/components/explicit/ExplicitText';
import OwnerBlock from '../../../src/newsfeed/activity/OwnerBlock';
import Actions from '../../../src/newsfeed/activity/Actions';
import ActivityMetrics from '../../../src/newsfeed/activity/metrics/ActivityMetrics';

import MediaView from '../../../src/common/components/MediaView';

import ActivityModel from '../../../src/newsfeed/ActivityModel';
import MindsVideo from '../../../src/media/MindsVideo';
import MindsVideoV2 from '../../../src/media/v2/mindsVideo/MindsVideo';

jest.mock('../../../src/media/MindsVideo', () => 'MindsVideo');
jest.mock('../../../src/media/v2/mindsVideo/MindsVideo', () => 'MindsVideoV2');
jest.mock('../../../src/common/components/Translate', () => 'Translate');
jest.mock(
  '../../../src/common/components/explicit/ExplicitText',
  () => 'ExplicitText',
);
jest.mock('../../../src/newsfeed/activity/OwnerBlock', () => 'OwnerBlock');
jest.mock('../../../src/newsfeed/activity/Actions', () => 'Actions');

jest.mock(
  '../../../src/newsfeed/activity/RemindOwnerBlock',
  () => 'RemindOwnerBlock',
);
jest.mock(
  '../../../src/newsfeed/activity/ActivityActionSheet',
  () => 'ActivityActionSheet',
);
jest.mock(
  '../../../src/newsfeed/activity/metrics/ActivityMetrics',
  () => 'ActivityMetrics',
);
jest.mock('../../../AppStores');

describe('Activity component', () => {
  let user, comments, entity, screen;
  beforeEach(() => {
    const navigation = { navigate: jest.fn(), push: jest.fn() };
    let activityResponse = activitiesServiceFaker().load(1);

    const model = ActivityModel.create(activityResponse.activities[0]);

    screen = shallow(<Activity entity={model} navigation={navigation} />);
  });

  it('renders correctly', async () => {
    screen.update();
    expect(screen).toMatchSnapshot();
  });

  it('should have a Media View', async () => {
    screen.update();

    expect(screen.find(MediaView)).toHaveLength(1);
  });

  it('should have the expectedComponents', async () => {
    screen.update();

    expect(screen.find(MediaView)).toHaveLength(1);
    expect(screen.find(Actions)).toHaveLength(1);
    expect(screen.find(OwnerBlock)).toHaveLength(1);
    expect(screen.find(ExplicitText)).toHaveLength(1);
    // expect(screen.find(ActivityMetrics)).toHaveLength(1);
    expect(screen.find(TouchableOpacity)).toHaveLength(1);
  });

  // it('should navToActivity on press', () => {
  //   // screen.update();
  //   let instance = screen.instance();
  //   const spy = jest.spyOn(instance, 'navToActivity');
  //   screen.find(TouchableOpacity).forEach((child) => {
  //     child.simulate('press');
  //   });

  //   expect(spy).toHaveBeenCalled();
  // });
});
