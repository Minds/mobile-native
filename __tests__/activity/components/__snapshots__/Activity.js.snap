// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Activity component renders correctly 1`] = `
<TouchableOpacity
  activeOpacity={0.8}
  delayPressIn={60}
  onLayout={[Function]}
  onLongPress={[Function]}
  onPress={[Function]}
  style={
    Array [
      Object {
        "overflow": "visible",
      },
      undefined,
      undefined,
      undefined,
    ]
  }
  testID="ActivityView"
>
  <Pinned
    entity={
      Object {
        "allow_comments": true,
        "attachment_guid": false,
        "blurb": false,
        "comments:count": undefined,
        "containerObj": undefined,
        "container_guid": "activityguid0",
        "custom_data": false,
        "custom_type": false,
        "description": "Congratulations! ",
        "edited": "",
        "entity_guid": "",
        "getLockType": [Function],
        "guid": "activityguid0",
        "impressions": 0,
        "isOwner": [Function],
        "is_visible": false,
        "mature": false,
        "mature_visibility": false,
        "message": "Message",
        "ownerObj": Object {
          "__list": null,
          "allow_comments": true,
          "blocked": undefined,
          "confirmEmail": [Function],
          "email_confirmed": false,
          "eth_wallet": "",
          "founder": false,
          "groupsCount": 0,
          "guid": "824853017709780997",
          "impressions": undefined,
          "isOwner": [Function],
          "is_admin": false,
          "last_accepted_tos": 0,
          "mature": false,
          "mature_visibility": false,
          "mode": 0,
          "onchain_booster": 0,
          "pending_subscribe": false,
          "permissions": Object {},
          "plus": false,
          "pro": false,
          "rewards": false,
          "subscribed": undefined,
          "subscribers_count": undefined,
          "subscriptions_count": 0,
          "subtype": false,
          "tags": Array [],
          "time_created": "1522036284",
          "type": "user",
          "urn": "",
          "username": "",
          "verified": false,
          "wire_rewards": undefined,
        },
        "owner_guid": "824853017709780997",
        "parent_guid": "838106762591510528",
        "paywall": "",
        "paywall_unlocked": false,
        "perma_url": false,
        "permissions": Object {},
        "pinned": false,
        "reminds": 0,
        "rowKey": "something0",
        "shouldBeBlured": [MockFunction] {
          "calls": Array [
            Array [],
          ],
          "results": Array [
            Object {
              "type": "return",
              "value": undefined,
            },
          ],
        },
        "subtype": "",
        "thumbnail_src": false,
        "thumbs:down:count": undefined,
        "thumbs:down:user_guids": undefined,
        "thumbs:up:count": undefined,
        "thumbs:up:user_guids": undefined,
        "time_created": "1522036284",
        "title": "TITLE",
        "type": "activity",
        "urn": "",
        "username": "",
        "wire_totals": Object {
          "tokens": 1000000000000000000,
        },
      }
    }
  />
  <OwnerBlock
    entity={
      Object {
        "allow_comments": true,
        "attachment_guid": false,
        "blurb": false,
        "comments:count": undefined,
        "containerObj": undefined,
        "container_guid": "activityguid0",
        "custom_data": false,
        "custom_type": false,
        "description": "Congratulations! ",
        "edited": "",
        "entity_guid": "",
        "getLockType": [Function],
        "guid": "activityguid0",
        "impressions": 0,
        "isOwner": [Function],
        "is_visible": false,
        "mature": false,
        "mature_visibility": false,
        "message": "Message",
        "ownerObj": Object {
          "__list": null,
          "allow_comments": true,
          "blocked": undefined,
          "confirmEmail": [Function],
          "email_confirmed": false,
          "eth_wallet": "",
          "founder": false,
          "groupsCount": 0,
          "guid": "824853017709780997",
          "impressions": undefined,
          "isOwner": [Function],
          "is_admin": false,
          "last_accepted_tos": 0,
          "mature": false,
          "mature_visibility": false,
          "mode": 0,
          "onchain_booster": 0,
          "pending_subscribe": false,
          "permissions": Object {},
          "plus": false,
          "pro": false,
          "rewards": false,
          "subscribed": undefined,
          "subscribers_count": undefined,
          "subscriptions_count": 0,
          "subtype": false,
          "tags": Array [],
          "time_created": "1522036284",
          "type": "user",
          "urn": "",
          "username": "",
          "verified": false,
          "wire_rewards": undefined,
        },
        "owner_guid": "824853017709780997",
        "parent_guid": "838106762591510528",
        "paywall": "",
        "paywall_unlocked": false,
        "perma_url": false,
        "permissions": Object {},
        "pinned": false,
        "reminds": 0,
        "rowKey": "something0",
        "shouldBeBlured": [MockFunction] {
          "calls": Array [
            Array [],
          ],
          "results": Array [
            Object {
              "type": "return",
              "value": undefined,
            },
          ],
        },
        "subtype": "",
        "thumbnail_src": false,
        "thumbs:down:count": undefined,
        "thumbs:down:user_guids": undefined,
        "thumbs:up:count": undefined,
        "thumbs:up:user_guids": undefined,
        "time_created": "1522036284",
        "title": "TITLE",
        "type": "activity",
        "urn": "",
        "username": "",
        "wire_totals": Object {
          "tokens": 1000000000000000000,
        },
      }
    }
    navigation={
      Object {
        "navigate": [MockFunction],
        "push": [MockFunction],
      }
    }
    rightToolbar={
      <View>
        <ActivityActionSheet
          entity={
            Object {
              "allow_comments": true,
              "attachment_guid": false,
              "blurb": false,
              "comments:count": undefined,
              "containerObj": undefined,
              "container_guid": "activityguid0",
              "custom_data": false,
              "custom_type": false,
              "description": "Congratulations! ",
              "edited": "",
              "entity_guid": "",
              "getLockType": [Function],
              "guid": "activityguid0",
              "impressions": 0,
              "isOwner": [Function],
              "is_visible": false,
              "mature": false,
              "mature_visibility": false,
              "message": "Message",
              "ownerObj": Object {
                "__list": null,
                "allow_comments": true,
                "blocked": undefined,
                "confirmEmail": [Function],
                "email_confirmed": false,
                "eth_wallet": "",
                "founder": false,
                "groupsCount": 0,
                "guid": "824853017709780997",
                "impressions": undefined,
                "isOwner": [Function],
                "is_admin": false,
                "last_accepted_tos": 0,
                "mature": false,
                "mature_visibility": false,
                "mode": 0,
                "onchain_booster": 0,
                "pending_subscribe": false,
                "permissions": Object {},
                "plus": false,
                "pro": false,
                "rewards": false,
                "subscribed": undefined,
                "subscribers_count": undefined,
                "subscriptions_count": 0,
                "subtype": false,
                "tags": Array [],
                "time_created": "1522036284",
                "type": "user",
                "urn": "",
                "username": "",
                "verified": false,
                "wire_rewards": undefined,
              },
              "owner_guid": "824853017709780997",
              "parent_guid": "838106762591510528",
              "paywall": "",
              "paywall_unlocked": false,
              "perma_url": false,
              "permissions": Object {},
              "pinned": false,
              "reminds": 0,
              "rowKey": "something0",
              "shouldBeBlured": [MockFunction] {
                "calls": Array [
                  Array [],
                ],
                "results": Array [
                  Object {
                    "type": "return",
                    "value": undefined,
                  },
                ],
              },
              "subtype": "",
              "thumbnail_src": false,
              "thumbs:down:count": undefined,
              "thumbs:down:user_guids": undefined,
              "thumbs:up:count": undefined,
              "thumbs:up:user_guids": undefined,
              "time_created": "1522036284",
              "title": "TITLE",
              "type": "activity",
              "urn": "",
              "username": "",
              "wire_totals": Object {
                "tokens": 1000000000000000000,
              },
            }
          }
          navigation={
            Object {
              "navigate": [MockFunction],
              "push": [MockFunction],
            }
          }
          onTranslate={[Function]}
          testID=""
        />
      </View>
    }
  >
    <View>
      <Text
        style={
          Array [
            Object {
              "color": "#888",
              "fontSize": 14,
            },
            Object {
              "paddingRight": 5,
            },
            undefined,
          ]
        }
      >
        3 years ago
      </Text>
    </View>
  </OwnerBlock>
  <View
    style={
      Object {
        "justifyContent": "center",
        "minHeight": 150,
      }
    }
  >
    <MediaView
      entity={
        Object {
          "allow_comments": true,
          "attachment_guid": false,
          "blurb": false,
          "comments:count": undefined,
          "containerObj": undefined,
          "container_guid": "activityguid0",
          "custom_data": false,
          "custom_type": false,
          "description": "Congratulations! ",
          "edited": "",
          "entity_guid": "",
          "getLockType": [Function],
          "guid": "activityguid0",
          "impressions": 0,
          "isOwner": [Function],
          "is_visible": false,
          "mature": false,
          "mature_visibility": false,
          "message": "Message",
          "ownerObj": Object {
            "__list": null,
            "allow_comments": true,
            "blocked": undefined,
            "confirmEmail": [Function],
            "email_confirmed": false,
            "eth_wallet": "",
            "founder": false,
            "groupsCount": 0,
            "guid": "824853017709780997",
            "impressions": undefined,
            "isOwner": [Function],
            "is_admin": false,
            "last_accepted_tos": 0,
            "mature": false,
            "mature_visibility": false,
            "mode": 0,
            "onchain_booster": 0,
            "pending_subscribe": false,
            "permissions": Object {},
            "plus": false,
            "pro": false,
            "rewards": false,
            "subscribed": undefined,
            "subscribers_count": undefined,
            "subscriptions_count": 0,
            "subtype": false,
            "tags": Array [],
            "time_created": "1522036284",
            "type": "user",
            "urn": "",
            "username": "",
            "verified": false,
            "wire_rewards": undefined,
          },
          "owner_guid": "824853017709780997",
          "parent_guid": "838106762591510528",
          "paywall": "",
          "paywall_unlocked": false,
          "perma_url": false,
          "permissions": Object {},
          "pinned": false,
          "reminds": 0,
          "rowKey": "something0",
          "shouldBeBlured": [MockFunction] {
            "calls": Array [
              Array [],
            ],
            "results": Array [
              Object {
                "type": "return",
                "value": undefined,
              },
            ],
          },
          "subtype": "",
          "thumbnail_src": false,
          "thumbs:down:count": undefined,
          "thumbs:down:user_guids": undefined,
          "thumbs:up:count": undefined,
          "thumbs:up:user_guids": undefined,
          "time_created": "1522036284",
          "title": "TITLE",
          "type": "activity",
          "urn": "",
          "username": "",
          "wire_totals": Object {
            "tokens": 1000000000000000000,
          },
        }
      }
      onPress={[Function]}
      style={
        Object {
          "flex": 1,
        }
      }
      width={750}
    />
    <View
      style={
        Object {
          "paddingHorizontal": 20,
          "paddingVertical": 15,
        }
      }
    >
      <ExplicitText
        entity={
          Object {
            "allow_comments": true,
            "attachment_guid": false,
            "blurb": false,
            "comments:count": undefined,
            "containerObj": undefined,
            "container_guid": "activityguid0",
            "custom_data": false,
            "custom_type": false,
            "description": "Congratulations! ",
            "edited": "",
            "entity_guid": "",
            "getLockType": [Function],
            "guid": "activityguid0",
            "impressions": 0,
            "isOwner": [Function],
            "is_visible": false,
            "mature": false,
            "mature_visibility": false,
            "message": "Message",
            "ownerObj": Object {
              "__list": null,
              "allow_comments": true,
              "blocked": undefined,
              "confirmEmail": [Function],
              "email_confirmed": false,
              "eth_wallet": "",
              "founder": false,
              "groupsCount": 0,
              "guid": "824853017709780997",
              "impressions": undefined,
              "isOwner": [Function],
              "is_admin": false,
              "last_accepted_tos": 0,
              "mature": false,
              "mature_visibility": false,
              "mode": 0,
              "onchain_booster": 0,
              "pending_subscribe": false,
              "permissions": Object {},
              "plus": false,
              "pro": false,
              "rewards": false,
              "subscribed": undefined,
              "subscribers_count": undefined,
              "subscriptions_count": 0,
              "subtype": false,
              "tags": Array [],
              "time_created": "1522036284",
              "type": "user",
              "urn": "",
              "username": "",
              "verified": false,
              "wire_rewards": undefined,
            },
            "owner_guid": "824853017709780997",
            "parent_guid": "838106762591510528",
            "paywall": "",
            "paywall_unlocked": false,
            "perma_url": false,
            "permissions": Object {},
            "pinned": false,
            "reminds": 0,
            "rowKey": "something0",
            "shouldBeBlured": [MockFunction] {
              "calls": Array [
                Array [],
              ],
              "results": Array [
                Object {
                  "type": "return",
                  "value": undefined,
                },
              ],
            },
            "subtype": "",
            "thumbnail_src": false,
            "thumbs:down:count": undefined,
            "thumbs:down:user_guids": undefined,
            "thumbs:up:count": undefined,
            "thumbs:up:user_guids": undefined,
            "time_created": "1522036284",
            "title": "TITLE",
            "type": "activity",
            "urn": "",
            "username": "",
            "wire_totals": Object {
              "tokens": 1000000000000000000,
            },
          }
        }
        navigation={
          Object {
            "navigate": [MockFunction],
            "push": [MockFunction],
          }
        }
        style={
          Array [
            Object {
              "fontFamily": "Roboto",
              "fontSize": 16,
              "lineHeight": 24,
            },
            Array [
              undefined,
              undefined,
            ],
          ]
        }
      />
      <Translate
        entity={
          Object {
            "allow_comments": true,
            "attachment_guid": false,
            "blurb": false,
            "comments:count": undefined,
            "containerObj": undefined,
            "container_guid": "activityguid0",
            "custom_data": false,
            "custom_type": false,
            "description": "Congratulations! ",
            "edited": "",
            "entity_guid": "",
            "getLockType": [Function],
            "guid": "activityguid0",
            "impressions": 0,
            "isOwner": [Function],
            "is_visible": false,
            "mature": false,
            "mature_visibility": false,
            "message": "Message",
            "ownerObj": Object {
              "__list": null,
              "allow_comments": true,
              "blocked": undefined,
              "confirmEmail": [Function],
              "email_confirmed": false,
              "eth_wallet": "",
              "founder": false,
              "groupsCount": 0,
              "guid": "824853017709780997",
              "impressions": undefined,
              "isOwner": [Function],
              "is_admin": false,
              "last_accepted_tos": 0,
              "mature": false,
              "mature_visibility": false,
              "mode": 0,
              "onchain_booster": 0,
              "pending_subscribe": false,
              "permissions": Object {},
              "plus": false,
              "pro": false,
              "rewards": false,
              "subscribed": undefined,
              "subscribers_count": undefined,
              "subscriptions_count": 0,
              "subtype": false,
              "tags": Array [],
              "time_created": "1522036284",
              "type": "user",
              "urn": "",
              "username": "",
              "verified": false,
              "wire_rewards": undefined,
            },
            "owner_guid": "824853017709780997",
            "parent_guid": "838106762591510528",
            "paywall": "",
            "paywall_unlocked": false,
            "perma_url": false,
            "permissions": Object {},
            "pinned": false,
            "reminds": 0,
            "rowKey": "something0",
            "shouldBeBlured": [MockFunction] {
              "calls": Array [
                Array [],
              ],
              "results": Array [
                Object {
                  "type": "return",
                  "value": undefined,
                },
              ],
            },
            "subtype": "",
            "thumbnail_src": false,
            "thumbs:down:count": undefined,
            "thumbs:down:user_guids": undefined,
            "thumbs:up:count": undefined,
            "thumbs:up:user_guids": undefined,
            "time_created": "1522036284",
            "title": "TITLE",
            "type": "activity",
            "urn": "",
            "username": "",
            "wire_totals": Object {
              "tokens": 1000000000000000000,
            },
          }
        }
        style={
          Object {
            "fontFamily": "Roboto",
            "fontSize": 16,
            "lineHeight": 24,
          }
        }
      />
    </View>
  </View>
  <Actions
    entity={
      Object {
        "allow_comments": true,
        "attachment_guid": false,
        "blurb": false,
        "comments:count": undefined,
        "containerObj": undefined,
        "container_guid": "activityguid0",
        "custom_data": false,
        "custom_type": false,
        "description": "Congratulations! ",
        "edited": "",
        "entity_guid": "",
        "getLockType": [Function],
        "guid": "activityguid0",
        "impressions": 0,
        "isOwner": [Function],
        "is_visible": false,
        "mature": false,
        "mature_visibility": false,
        "message": "Message",
        "ownerObj": Object {
          "__list": null,
          "allow_comments": true,
          "blocked": undefined,
          "confirmEmail": [Function],
          "email_confirmed": false,
          "eth_wallet": "",
          "founder": false,
          "groupsCount": 0,
          "guid": "824853017709780997",
          "impressions": undefined,
          "isOwner": [Function],
          "is_admin": false,
          "last_accepted_tos": 0,
          "mature": false,
          "mature_visibility": false,
          "mode": 0,
          "onchain_booster": 0,
          "pending_subscribe": false,
          "permissions": Object {},
          "plus": false,
          "pro": false,
          "rewards": false,
          "subscribed": undefined,
          "subscribers_count": undefined,
          "subscriptions_count": 0,
          "subtype": false,
          "tags": Array [],
          "time_created": "1522036284",
          "type": "user",
          "urn": "",
          "username": "",
          "verified": false,
          "wire_rewards": undefined,
        },
        "owner_guid": "824853017709780997",
        "parent_guid": "838106762591510528",
        "paywall": "",
        "paywall_unlocked": false,
        "perma_url": false,
        "permissions": Object {},
        "pinned": false,
        "reminds": 0,
        "rowKey": "something0",
        "shouldBeBlured": [MockFunction] {
          "calls": Array [
            Array [],
          ],
          "results": Array [
            Object {
              "type": "return",
              "value": undefined,
            },
          ],
        },
        "subtype": "",
        "thumbnail_src": false,
        "thumbs:down:count": undefined,
        "thumbs:down:user_guids": undefined,
        "thumbs:up:count": undefined,
        "thumbs:up:user_guids": undefined,
        "time_created": "1522036284",
        "title": "TITLE",
        "type": "activity",
        "urn": "",
        "username": "",
        "wire_totals": Object {
          "tokens": 1000000000000000000,
        },
      }
    }
  />
</TouchableOpacity>
`;
